mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx

```diff
@@ -584,37 +593,47 @@ export const registerClickEventsForGraphElements = (graphOutput: GraphOutput, se
     xtitle.style.pointerEvents = 'all'
     ytitle.style.pointerEvents = 'all'
 
+    /**
+     * Open context menu when right clicked
+     */
+    gtitle.addEventListener('contextmenu', (e: any) => {
+        e.preventDefault();
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(gtitle, 'gtitle', graphOutput, 'context-menu'))
+    })
+    xtitle.addEventListener('contextmenu', (e: any) => {
+        e.preventDefault();
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(xtitle, 'xtitle', graphOutput, 'context-menu'))
+    })
+    ytitle.addEventListener('contextmenu', (e: any) => {
+        e.preventDefault();
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(ytitle, 'ytitle', graphOutput, 'context-menu'))
+    })
+
     /**
      * Set selected graph element when clicked
      */
     gtitle.addEventListener('click', () => {
-        setSelectedGraphElement({
-            element: 'gtitle',
-        })
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(xtitle, 'gtitle', graphOutput, 'none'))
     })
     xtitle.addEventListener('click', () => {
-        setSelectedGraphElement({
-            element: 'xtitle',
-        });
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(xtitle, 'xtitle', graphOutput, 'none'))
     })
     ytitle.addEventListener('click', () => {
-        setSelectedGraphElement({
-            element: 'ytitle',
-        })
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(ytitle, 'ytitle', graphOutput, 'none'))
     })
 
     /**
      * Open popup when double clicked
      */
     gtitle.addEventListener('dblclick', () => {
-        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(gtitle, 'gtitle', graphOutput))
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(gtitle, 'gtitle', graphOutput, 'editor'))
```

Comment:
maybe `popup-text-editor` or something more descriptive; editor is really vague,.